server-spi-private/src/main/java/org/keycloak/utils/CredentialHelper.java

```diff
@@ -90,6 +95,33 @@ public static ConfigurableAuthenticatorFactory getConfigurableAuthenticatorFacto
          return factory;
     }
 
+    public static RequiredActionFactory getConfigurableRequiredActionFactory(KeycloakSession session, String providerId) {
```

Comment:
This and the following helper functions should be moved elsewhere... just placed them here because the authenticator config methods were also located here.

Reply:
Nitpick: Can be ok to introduce new class like `RequiredActionHelper` and add those methods here as they don't have much to do with credentials...